docs/spec/spec.md

```diff
@@ -2758,9 +2763,215 @@ public function main() returns error? {
+
+###### Example: Federated Entity Definition and Corresponding GraphQL Schema
+
+<table>
+    <tr>
+        <th>Example</th>
+        <th>Ballerina definition</th>
+        <th>GraphQL schema</th>
+    </tr>
+    <tr>
+        <td>
+            Simple key
+        </td>
+        <td>
+            <pre lang="ballerina">
+```ballerina
+@subgraph:Entity {
+    key: "id",
+    resolveReference: resolveProduct
+}
+type Product record {
+    string id;
+    string name;
+    int price;
+};
+```
+            </pre>
+        </td>
+        <td>
+            <pre lang="graphql">
+```graphql
+type Product @key(fields: "id") {
+    id: String!
+    name: String!
+    price: Int!
+}
+```
+            </pre>
+        </td>
+    </tr>
+    <tr>
+        <td>
+            Mutiple keys
+        </td>
+        <td>
+            <pre lang="ballerina">
+```ballerina
+@subgraph:Entity {
+    key: ["id", "sku"],
+    resolveReference: resolveProduct
+}
+type Product record {
+    string id;
+    string name;
+    int price;
+};
+```
+            </pre>
+        </td>
+        <td>
+            <pre lang="graphql">
+```graphql
+type Product @key(fields: "id") @key(fields: "sku") {
+    id: String!
+    name: String!
+    price: Int!
+}
+```
+            </pre>
+        </td>
+    </tr>
+    <tr>
+        <td>
+            Compound key
+        </td>
+        <td>
+            <pre lang="ballerina">
+```ballerina
+@subgraph:Entity {
+    key: "id organization { id }",
+    resolveReference: resolveUser
+}
+type User record {
+    string id;
+    Organization organization;
+}
+```
+            </pre>
+        </td>
+        <td>
+            <pre lang="graphql">
+```graphql
+type User @key(fields: "id organization { id }") {
+    id: String!
+    organization: Organization!
+}
+```
+            </pre>
+        </td>
+    </tr>
+      <tr>
+        <td>
+            Non resolvable
+        </td>
+        <td>
+            <pre lang="ballerina">
+```ballerina
+@subgraph:Entity {
+    key: "id"
+    resolveReference: ()
+}
+type Product record {
+  id: String!
+}
+```
+            </pre>
+        </td>
+        <td>
+            <pre lang="graphql">
+```graphql
+type Product @key(fields: "id", resolvable: false) {
+    id: ID!
+}
+```
+            </pre>
+        </td>
+    </tr>
+</table>
+
+#### 13.1.3 The `subgraph:ReferenceResolver`
+
+Reference resolver is a function that resolves an entity of a specific type using its primary key. When the graph router needs to resolve an entity, it calls the reference resolver and passes the primary key and the `__typename` field of the entity. The reference resolver then returns the entity with the given primary key. Following is the type definition of a reference resolver defined in `graphql.subgraph` module.
```

Comment:
From an outsider's point of view, this information is not required, right? What they need to know is that the provided reference resolver will be executed when a particular entity is needed to be resolved. I.e., The reference resolver is the function that needs to be executed to resolve a given entity.

Reply:
updated